node/internal.ts

```diff
@@ -171,8 +171,27 @@ export function _setResourcePath(path: string): void {
 }
 
 /**
+ * Imports localized strings from json file.
+ * It can be used by common modules to load module specific loc strings
+ *
+ * @param     path      Full path to the json.
+ * @returns   void
+ */
+export function _importLocStrings(path: string): void {
```

Comment:
Yes. It's up to the module author. Even without a single predictable entry file for the module, each possible entry file can reference a common file that imports the resource strings. If two separate files require the same common file, node caches the require and it's only actually loaded once.

The other suggestion that was previously brought up was at build time in the vsts-tasks repo to inject lib.json messages into task.json. The problem I have with that approach is that we need to find a solution that works for customers too.

If we can find out the __dirname of the caller then we can automatically look for and load a task.json/lib.json without requiring anyone to call setResourcePath. I don't know whether that is feasible.